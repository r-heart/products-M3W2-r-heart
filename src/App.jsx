import ProductList from "./components/ProductList/ProductList.jsx";
import SearchBar from "./components/SearchBar";
import useFetch from "./hooks/useFetch";
import useSearch from "./hooks/useSearch.jsx";

const BASE_URL = "http://localhost:3001/products";

function App() {
  const products = useFetch(BASE_URL);
  const { setSearchTerm, setCategory, setInStockOnly, filteredProducts } =
    useSearch(products);

  return (
    <>
      <h1 className="my-4 text-center text-2xl font-black">Product App</h1>
      <SearchBar
        categories={[...new Set(products.map((product) => product.category))]}
        setSearchTerm={setSearchTerm}
        setCategory={setCategory}
        setInStockOnly={setInStockOnly}
      />
      <ProductList products={filteredProducts} />
    </>
  );
}
export default App;
