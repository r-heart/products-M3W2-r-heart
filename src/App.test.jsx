import { render, screen } from "@testing-library/react";
import { expect, test } from "vitest";
import App from "./App";

test("renders correctly", () => {
  const appRender = render(<App />);
  expect(appRender).toMatchSnapshot();
});
