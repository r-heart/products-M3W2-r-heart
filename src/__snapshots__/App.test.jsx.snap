// Vitest Snapshot v1

exports[`renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1
        class="my-4 text-center text-2xl font-black"
      >
        Product App
      </h1>
      <form
        class="my-4 flex flex-col items-center gap-y-4"
      >
        <label
          class="sr-only"
          for="search"
        >
          Search
        </label>
        <input
          class="border-gray-300mpx-4 focus-border-indigo-500 rounded-md py-2 shadow-sm focus:ring-indigo-500"
          id="search"
          placeholder="Search"
          type="search"
        />
        <label
          class="sr-only"
          for="category"
        >
          Category
        </label>
        <select
          id="category"
        >
          <option
            value="all"
          >
            All
          </option>
        </select>
        <div
          class="flex items-center gap-x-2"
        >
          <input
            id="in-stock-only"
            type="checkbox"
          />
          <label
            for="in-stock-only"
          >
            Only show products in stock
          </label>
        </div>
      </form>
      <ul
        class="container mx-auto mt-8 flex flex-col items-center justify-center"
      />
    </div>
  </body>,
  "container": <div>
    <h1
      class="my-4 text-center text-2xl font-black"
    >
      Product App
    </h1>
    <form
      class="my-4 flex flex-col items-center gap-y-4"
    >
      <label
        class="sr-only"
        for="search"
      >
        Search
      </label>
      <input
        class="border-gray-300mpx-4 focus-border-indigo-500 rounded-md py-2 shadow-sm focus:ring-indigo-500"
        id="search"
        placeholder="Search"
        type="search"
      />
      <label
        class="sr-only"
        for="category"
      >
        Category
      </label>
      <select
        id="category"
      >
        <option
          value="all"
        >
          All
        </option>
      </select>
      <div
        class="flex items-center gap-x-2"
      >
        <input
          id="in-stock-only"
          type="checkbox"
        />
        <label
          for="in-stock-only"
        >
          Only show products in stock
        </label>
      </div>
    </form>
    <ul
      class="container mx-auto mt-8 flex flex-col items-center justify-center"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
